Lib/os.py

```diff
@@ -826,6 +830,29 @@ def decode(value):
 del _create_environ_mapping
 
 
+if _exists("_get_user_default_environ"):
+    def get_user_default_environ():
+        """
+        Get the default environment of the current process user as a dictionary.
+        """
+        env_str = _get_user_default_environ()
+        env = {}
+        for entry in env_str.split('\0'):
+            parts = entry.split('=', 1)
+            if len(parts) != 2:
+                # Silently skip variable with no name
+                continue
+
+            name, value = parts
+            if not name:
+                # Silently skip variable with empty name
+                continue
+
+            # If a variable is set twice, use the first value
+            env.setdefault(name, value)
```

Comment:
I suppose this doesn't hurt. But it shouldn't be possible to get duplicates unless there's a serious bug. The new environment is built by assigning values to the environment block, which is handled as a parameter by the same code that implements WinAPI `SetEnvironmentVariableW()`.